model/incoming_webhook.go

```diff
@@ -9,25 +9,32 @@ import (
 	"io"
 	"net/http"
 	"regexp"
+	"strconv"
 )
 
 const (
 	DEFAULT_WEBHOOK_USERNAME = "webhook"
 )
 
 type IncomingWebhook struct {
-	Id            string `json:"id"`
-	CreateAt      int64  `json:"create_at"`
-	UpdateAt      int64  `json:"update_at"`
-	DeleteAt      int64  `json:"delete_at"`
-	UserId        string `json:"user_id"`
-	ChannelId     string `json:"channel_id"`
-	TeamId        string `json:"team_id"`
-	DisplayName   string `json:"display_name"`
-	Description   string `json:"description"`
-	Username      string `json:"username"`
-	IconURL       string `json:"icon_url"`
-	ChannelLocked bool   `json:"channel_locked"`
+	Id                 string          `json:"id"`
+	CreateAt           int64           `json:"create_at"`
+	UpdateAt           int64           `json:"update_at"`
+	DeleteAt           int64           `json:"delete_at"`
+	UserId             string          `json:"user_id"`
+	ChannelId          string          `json:"channel_id"`
+	TeamId             string          `json:"team_id"`
+	DisplayName        string          `json:"display_name"`
+	Description        string          `json:"description"`
+	Username           string          `json:"username"`
+	IconURL            string          `json:"icon_url"`
+	ChannelLocked      bool            `json:"channel_locked"`
+	SecretToken        string          `json:"secret_token"`
```

Comment:
New fields here that are persisted to the database should be added to https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/upgrade.go so that existing database schemas get the new fields. It's not there yet, but the migration function would be `upgradeDatabaseToVersion522` given this changes would be targeting v5.22.

Reply:
Thanks for point this out. As of now, I used 521.

Reply:
Apologies for the ambiguity on my part -- v5.21 is our upcoming quality release, which contains no new features. So the earliest your changes could be included is our v5.22 (April) release, if merged to `master` prior to March 18th and our feature cutoff for v5.22. Thereafter, it would be targeted at v5.24 (June).